torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -200,13 +201,14 @@ def unshard(self, async_op: bool = False):
             # used in the all-gather streams
             self._wait_all_gather_streams_on_event(self._reshard_after_forward_event)
             self._reshard_after_forward_event = None
-        self._all_gather_result = foreach_all_gather(
-            self.fsdp_params,
-            self._all_gather_process_group,
-            async_op,
-            *self.comm_ctx.get_all_gather_streams(self._training_state),
-            self.device,
-        )
+        with record_function(self._with_fqn("FSDP::all_gather")):
```

Comment:
This would cover both the all-gather copy-in and the all-gather itself. Since all-gather copy-in is now its own custom op, it the trace would look something like this:
![Screenshot 2024-06-13 at 12 59 13 PM](https://github.com/pytorch/pytorch/assets/31054793/b16e100b-a7ec-4f22-b17e-1055b8df1c1b)

Personally, I think it is not too confusing.